src/main/java/dao/TicketDao.java

```diff
@@ -0,0 +1,86 @@
+    private static final String FETCH_TICKET_BY_USER_ID = "SELECT ticket_id, user_id, ticket_type, creation_date FROM tickets WHERE user_id = ?";
+    private static final String UPDATE_TICKET_TYPE = "UPDATE tickets SET ticket_type = ?::ticket_type WHERE id = ?";
+
+    public void saveTicket(Ticket ticket, int userId) {
+        try {
+            connection = dataSource.getConnection();
+            preparedStatement = connection.prepareStatement(SAVE_TICKET);
+            preparedStatement.setInt(1, ticket.getId());
+            preparedStatement.setInt(2, userId);
+            preparedStatement.setString(3, ticket.getTicketType().name());
+            preparedStatement.setTimestamp(4, Timestamp.valueOf(ticket.getTicketCreationTime()));
+            System.out.println(preparedStatement.executeUpdate());
+            preparedStatement.close();
+            connection.close();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public Ticket fetchTicketById(int ticketId) {
+        try {
+            connection = dataSource.getConnection();
+            preparedStatement = connection.prepareStatement(FETCH_TICKET_BY_ID);
+            preparedStatement.setInt(1, ticketId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                Ticket ticket = new Ticket();
+                ticket.setId(resultSet.getInt(1));
+                ticket.setTicketType(TicketType.valueOf(resultSet.getString(3)));
+                ticket.setTime(resultSet.getTimestamp(4).toLocalDateTime());
+                return ticket;
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+        return null;
+    }
+
+    public List<Ticket> fetchTicketByUserId(int userId) {
+        try {
+            connection = dataSource.getConnection();
+            preparedStatement = connection.prepareStatement(FETCH_TICKET_BY_USER_ID);
+            preparedStatement.setInt(1, userId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Ticket> tickets = new ArrayList<>();
+            while (resultSet.next()) {
+                Ticket ticket = new Ticket();
+                ticket.setId(resultSet.getInt(1));
+                ticket.setTicketType(TicketType.valueOf(resultSet.getString(3)));
+                ticket.setTime(resultSet.getTimestamp(4).toLocalDateTime());
+                tickets.add(ticket);
+            }
+            return tickets;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void updateTicketType(int ticketId, TicketType ticketType) {
+        try {
+            connection = dataSource.getConnection();
+            preparedStatement = connection.prepareStatement(UPDATE_TICKET_TYPE);
+            preparedStatement.setString(1, ticketType.name());
+            preparedStatement.setInt(2, ticketId);
```

Comment:
it doesn't work, because:
1. check your query UPDATE_TICKET_TYPE. there is no such column `id` into `tickets` table
2. missed last command: preparedStatement.executeUpdate();

Reply:
Fix in https://github.com/kolosokvit/ticket-service/pull/4/commits/8707b6a9b9ce6c4abcf171dbfcdacc588b249bc3